File: 백준/Silver/11060. 점프 점프/점프 점프.java
import java.util.Scanner;
import java.util.Arrays;

public class Main {
    public static void main(String args[]) {
      Scanner s = new Scanner(System.in);
      int n = s.nextInt();
      int[] jump = new int[1001];
      int[] dp = new int[1001];
      
      for(int i=1;i<=n;i++)
      {
          jump[i] = s.nextInt();
          dp[i] = 987654321;
      }
      dp[1] = 0;
      for(int i=1;i<=n;i++)
      {
          for(int j=0;j<=jump[i];j++) if((i+j)<=n) dp[i+j] = Math.min(dp[i]+1,dp[i+j]);
      }
      if(dp[n] == 987654321) System.out.print(-1);
      else System.out.print(dp[n]);
    }
}

File: 백준/Silver/11060. 점프 점프/점프 점프.kt
import java.util.*


fun main(args: Array<String>) {
    val s= Scanner(System.`in`)
    val n = s.nextInt()
    var jump = Array(1001) {0}
    var dp = Array(1001) {0}
    
    for(i in 1..n)
    {
        jump[i] = s.nextInt()
        dp[i] = 987654321
    }
    dp[1] = 0
    for(i in 1..n)
    {
        for(j in 0..jump[i])
        {
            if((i+j)<=n) dp[i+j] = Math.min(dp[i]+1,dp[i+j])
        }
    }
    if(dp[n]==987654321) print(-1)
    else print(dp[n])
}